synapsis/src/main/java/com/soen/synapsis/appuser/jobposting/JobPosting.java

```diff
@@ -0,0 +1,90 @@
+package com.soen.synapsis.appuser.jobposting;
+
+import com.soen.synapsis.appuser.AppUser;
+import com.soen.synapsis.appuser.Role;
+
+public class JobPosting {
+
+    private final AppUser creator;
+    private String position;
+    private String company;
+    private String address;
+    private String description;
+    private JobType type;
+    private int numAvailable;
+    private int numApplicants;
+
+    public JobPosting (AppUser creator, String position, String company, String address, String description, JobType type, int numAvailable, int numApplicants) {
+        if (creator.getRole() != Role.RECRUITER) {
```

Comment:
I don't think this sort of business logic should be in the data model class. It seems more like something that belongs in a service layer.